plugin/session_buffer.py

```diff
@@ -611,6 +619,40 @@ def set_semantic_tokens_pending_refresh(self, needs_refresh: bool = True) -> Non
     def get_semantic_tokens(self) -> List[SemanticToken]:
         return self.semantic_tokens.tokens
 
+    # --- textDocument/inlayHint ----------------------------------------------------------------------------------
+
+    def do_inlay_hints_async(self, view: sublime.View) -> None:
+        if not userprefs().show_inlay_hints:
+            return
+        if not self.session.has_capability("inlayHintProvider"):
+            return
+        params = {
+            "textDocument": text_document_identifier(view),
+            "range": entire_content_range(view).to_lsp()
+        }  # type: InlayHintParams
+        self.session.send_request_async(Request.inlayHint(params, view), self._on_inlay_hints_async)
+
+    def _on_inlay_hints_async(self, response: InlayHintResponse) -> None:
+        if response:
+            view = self.some_view()
+            if not view:
+                return
+            phantoms = [inlay_hint_to_phantom(view, inlay_hint, self.session) for inlay_hint in response]
+            sublime.set_timeout(lambda: self.present_inlay_hints_async(phantoms))
```

Comment:
Not entirely sure how a None/null response should be interpreted but I would think that it might be correct to clear phantoms in that case. Maybe some servers instead of sending empty array would send null instead to indicate that no hints are to be shown.